### PR TITLE
Add astc-codec/20190617

### PR DESCRIPTION
**Pull request by intelligide**
Specify library name and version:  **astc-codec/cci.20190617**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/astc-codec/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+class AstcCodecConan(ConanFile):
+    name = "astc-codec"
+    description = " A software ASTC decoder implementation which supports the ASTC LDR profile"
+    homepage = "https://github.com/google/astc-codec"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "Apache-2.0"
+    topics = ("conan", "astc", "codec")
+    settings = "os", "compiler", "arch", "build_type"
+    options = {
+        "fPIC": [True, False],
```
Comment (uilianries) [inline recipes/astc-codec/all/conanfile.py:15]: Why there is no shared option? I see no restriction: https://github.com/google/astc-codec/blob/master/src/decoder/CMakeLists.txt#L17

File: recipes/astc-codec/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
```
Comment (uilianries) [inline recipes/astc-codec/all/conanfile.py:35]:
```suggestion
            del self.options.fPIC

    def configure(self):
        if self.settings.compiler.cppstd:
            tools.check_min_cppstd(self, "11")
```

It requires C++11

File: recipes/astc-codec/all/conanfile.py (hunk 1 above)
Comment (intelligide) [reply recipes/astc-codec/all/conanfile.py:15]: The library doesn't export symbols. And `CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS` introduces some linking errors.

Comment (uilianries) [reply recipes/astc-codec/all/conanfile.py:15]: Okay, I didn't find any reference about this error, but I can reproduce it, so I've opened an issue to the author: https://github.com/google/astc-codec/issues/14

File: recipes/astc-codec/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
+        self.copy("*.h", dst="include", src=os.path.join(self._source_subfolder, "include"))
+        self.copy("*.lib", src=os.path.join(self._build_subfolder, "lib"), dst="lib", keep_path=False)
+        self.copy("*.dll", src=os.path.join(self._build_subfolder, "bin"), dst="bin", keep_path=False)
+        self.copy("*.exe", src=os.path.join(self._build_subfolder, "bin"), dst="bin", keep_path=False)
```
Comment (uilianries) [inline recipes/astc-codec/all/conanfile.py:63]: astc_inspector_cli will be copied, but is not listed in env_info.PATH.

**Review by danimtb — APPROVED** [review]

**Pull request merged**